pkg/operator/defaultstorageclass/controller.go

```diff
@@ -0,0 +1,182 @@
+package defaultstorageclass
+
+import (
+	"context"
+	"errors"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorapi "github.com/openshift/api/operator/v1"
+	cfginformers "github.com/openshift/client-go/config/informers/externalversions"
+	openshiftv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/cluster-storage-operator/pkg/generated"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	storagev1 "k8s.io/api/storage/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	errutil "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/listers/storage/v1"
+	"k8s.io/klog"
+)
+
+const (
+	conditionsPrefix      = "DefaultStorageClassController"
+	infraConfigName       = "cluster"
+	disabledConditionType = "Disabled"
+)
+
+var unsupportedPlatformError = errors.New("unsupported platform")
+
+// This Controller deploys a default StorageClass for in-tree volume plugins,
+// based on the underlying cloud (read from Infrastructure instance).
+// It produces following Conditions:
+// DefaultStorageClassControllerAvailable: the default storage class has been
+//    created.
+// DefaultStorageClassControllerProgressing - the default storage class has
+//    not been created yet (typically on error).
+// DefaultStorageClassControllerDegraded - error creating the storage class.
+type Controller struct {
+	operatorClient     v1helpers.OperatorClient
+	kubeClient         kubernetes.Interface
+	infraLister        openshiftv1.InfrastructureLister
+	storageClassLister v1.StorageClassLister
+	eventRecorder      events.Recorder
+}
+
+func NewController(
+	operatorClient v1helpers.OperatorClient,
+	kubeClient kubernetes.Interface,
+	kubeInformers informers.SharedInformerFactory,
+	infraInformer cfginformers.SharedInformerFactory,
+	eventRecorder events.Recorder) factory.Controller {
+	c := &Controller{
+		operatorClient:     operatorClient,
+		kubeClient:         kubeClient,
+		infraLister:        infraInformer.Config().V1().Infrastructures().Lister(),
+		storageClassLister: kubeInformers.Storage().V1().StorageClasses().Lister(),
+		eventRecorder:      eventRecorder,
+	}
+	return factory.New().WithSync(c.sync).WithSyncDegradedOnError(operatorClient).WithInformers(
+		operatorClient.Informer(),
+		infraInformer.Config().V1().Infrastructures().Informer(),
+		kubeInformers.Storage().V1().StorageClasses().Informer(),
+	).ToController("DefaultStorageClassController", eventRecorder)
```

Comment:
Do we need to also call `ResyncEver` here? From docs:

>// ResyncEvery will cause the Sync() function to be called periodically, regardless of informers.

Reply:
Each informer already has 20 minutes resync interval.